### ISSUE TITLE
Grafana show context support

### ISSUE THREAD

**Issue by clrxbl** [VictoriaMetrics/victorialogs-datasource#41]
This is a feature suggestion. Grafana Loki allows you to show the context, so the log lines before/after a specific log line you click on. It would be very useful if VictoriaLogs could do this aswell.

Example:
![CleanShot 2024-07-05 at 02 50 17@2x](https://github.com/VictoriaMetrics/victorialogs-datasource/assets/28601081/a2cead37-beaf-43ba-b887-fc62576a44a5)

![CleanShot 2024-07-05 at 02 51 22@2x](https://github.com/VictoriaMetrics/victorialogs-datasource/assets/28601081/f7afa38e-1f22-4171-acc4-855f40d614b5)

**Comment by hagen1778** [comment]
Can be implemented via https://docs.victoriametrics.com/victorialogs/logsql/#stream_context-pipe

**Comment by Andrey36652** [comment]
I'm +1 on this feature. I have very little knowledge about plugin development for grafana. How hard it will be to parse existing user query and inject stream context pipe in it like @hagen1778 suggested?

**Comment by worker24h** [comment]
The functions of logs pannel, if  implement show-context , we need to change log pannel code of grafana.

**Comment by hagen1778** [comment]
@dmitryk-dk @Loori-R I think this enhancement requires changes on both: backend and frontend.

**Comment by worker24h** [comment]
> @dmitryk-dk @Loori-R I think this enhancement requires changes on both: backend and frontend.

No, it is only frontend.  I can commit a PR for this, I am developing it.

**Comment by dmitryk-dk** [comment]
Hi @clrxbl ! The feature was added in the [v0.16.0](https://github.com/VictoriaMetrics/victorialogs-datasource/releases/tag/v0.16.0) release.

**Comment by mitsurugi-dev** [comment]
@dmitryk-dk Hello, please tell me, it says that the Log context tree was added to VictoriaLogs as in Loki, but I use victoriametrics-logs-datasource @ 0.26.3 and it does not display anything there. Has this functionality been added to VictoriaLogs or not?

**Comment by arturminchukov** [comment]
@mitsurugi-dev 
VictoriaLogs datasource has the context feature. You can find the `show context` button in the log line:

<img width="1523" height="1326" alt="Image" src="https://github.com/user-attachments/assets/7601d262-afe3-4d24-b33f-5a549ac7e9fb" />

You can try it on our playground: https://play-grafana.victoriametrics.com/explore 
If you mean something different, let me know.

**Comment by mitsurugi-dev** [comment]
@arturminchukov It does not display the label as in Loki

<img width="2134" height="499" alt="Image" src="https://github.com/user-attachments/assets/40a5f658-19da-420a-a977-768f51793acc" />

The second photo shows an example of Loki.

<img width="814" height="214" alt="Image" src="https://github.com/user-attachments/assets/0ded05f0-ca1e-4837-bd38-474a705b22b2" />

**Comment by mitsurugi-dev** [comment]
@arturminchukov Should these fields be displayed in Victorialogs?

**Comment by arturminchukov** [comment]
No, right now, we don't have such a feature. I created an issue  https://github.com/VictoriaMetrics/victorialogs-datasource/issues/665 to add it.
